### DISCUSSION TITLE
关于[边缘滚动-右侧 亮度调节]功能的疑问

### DISCUSSION THREAD

**Discussion by HorrZzz** [shuax/MouseInc.Settings#300] [Ideas]
# 疑问:
想问下 亮度调节是通过什么原理来调节的?我发现通过MouseInc调节亮度后,并非是调节的显示器的亮度.

# 建议:
能否设置一个开关,用来支持DDC/CI协议控制显示器的亮度?

**Comment by Ctrl-Ling** [comment]
最新版的亮度调节是使用WMI协议进行硬件亮度调节的，DDC/CI协议目前应该还不支持；
以前的旧版本应该是利用遮罩进行软件层的伪亮度调节的；

其实我更希望能够让旧版本的利用遮罩降低亮度的方式回归，然后增加开关控制使用哪种方式进行亮度条件，以便在不支持WMI调节的显示器上起码能用遮罩进行控制，希望作者可以看到我的建议~

另外，微软商店中的twinkletray软件可以实现DDC/CI协议控制，楼主可以先用着

**Reply by HorrZzz** [reply]
感谢您的细心解答,目前正在试用twinkletray软件进行DDC/CI协议控制.
还有有个问题想讨论先,目前市面上的显示器均支持了DDC/CI协议了吧?我有一台很老的三星显示器,都支持DDC/CI协议.

**Reply by Ctrl-Ling** [reply]
>另外，你可以将mouseInc的亮度调节映射成 twinkletray 的亮度调节按键快捷方式使用，效果拔群

协议这个不好说的，我有台很老的显示器，反倒是支持WMI调节方式。大部分笔记本电脑屏幕现在也是用的WMI协议；只能寄希望于作者大大适配DDC/CI吧，也不能强求，因为需要硬件调试

**Reply by HorrZzz** [reply]
> > 另外，你可以将mouseInc的亮度调节映射成 twinkletray 的亮度调节按键快捷方式使用，效果拔群
> 
> 协议这个不好说的，我有台很老的显示器，反倒是支持WMI调节方式。大部分笔记本电脑屏幕现在也是用的WMI协议；只能寄希望于作者大大适配DDC/CI吧，也不能强求，因为需要硬件调试

"将mouseInc的亮度调节映射成 twinkletray 的亮度调节按键快捷方式使用" 确实是个好办法,感谢!

**Comment by shuax** [comment]
默认是WMI，就是系统自带的那种方式，不支持的时候会切换成最老的伽马调节。DDC/CI我没有支持的显示器。

**Reply by HorrZzz** [reply]
> 默认是WMI，就是系统自带的那种方式，不支持的时候会切换成最老的伽马调节。DDC/CI我没有支持的显示器。

好的,我现在使用 @Ctrl-Ling 提供的快捷键映射twinkletray的方法实现DDC/CI协议控制 也挺好的.

**Reply by Ctrl-Ling** [reply]
那我这边的情况可能是有Bug或者说是不能正确识别显示器

我这边的场景是这样的：笔记本+外接DDC/CI显示器，笔记本屏幕不使用，滑动调节亮度，显示器没反应，估计是识别到笔记本自带的显示器用了WMI而没切成伽马调节的方式？
要不在MouseInc3加一个方式的选择？

亮度调节方式：默认（自动识别）/WMI/伽马调节，如果不生效我还可以手动选择伽马调节

**Reply by shuax** [reply]
因为只处理了主屏幕，没支持多显示器。
